tf_quant_finance/black_scholes/approximations/american_option.py

```diff
@@ -601,5 +681,63 @@ def _npdf(x):
   return tf.math.exp(-0.5 * x**2) / _SQRT_2_PI
 
 
+def _cbnd(dh, dk, rho):
+  """Compute values for the cumulative standard bivariate normal distribution,
+     specifically for use with Bjerksund-Stensland 2002 which always gives
+     the same value for rho.
+  #### References:
+  [1] Genz, A., Numerical Computation of Rectanglar Bivariate and Trivariate
+        Normal and t Probabilities, 2004
+    http://www.math.wsu.edu/faculty/genz/papers/bvnt.pdf"""
+
+  # Python code adapted from Fortran code and naming convention will align
+  # variables with the variables named in BVND function.
+  # http://www.math.wsu.edu/faculty/genz/software/fort77/tvpack.f
+
+  dtype = rho.dtype
+
+  # Gauss Legendre Points and Weights
+  w = tf.constant([0.01761400713915212, 0.04060142980038694, 0.06267204833410906,
+                   0.08327674157670475, 0.1019301198172404, 0.1181945319615184,
+                   0.1316886384491766, 0.1420961093183821, 0.1491729864726037,
+                   0.1527533871307259], dtype=dtype)
+
+  x = tf.constant([-0.9931285991850949, -0.9639719272779138, -0.9122344282513259,
+                   -0.8391169718222188, -0.7463319064601508, -0.6360536807265150,
+                   -0.5108670019508271, -0.3737060887154196, -0.2277858511416451,
+                   -0.07652652113349733], dtype=dtype)
+
+  h = tf.cast(-dh, dtype=dtype)
+  k = tf.cast(-dk, dtype=dtype)
+  hk = h * k
+
+  # In order to avoid `shape_invariants` argument in the while_loop, set
+  # `bvn` to be a Tensor of zeros of the appropriate shape
+  bvn = tf.zeros_like(hk)
+
+  hs = (h * h + k * k) / 2
+  asr = tf.math.asin(rho)
+
+  def batched_calc(hk, hs, asr, x, w):
+      for _ in range(hk.shape.rank):
```

Comment:
could we maybe have any better name for this function? Does it correspond to anything from the paper/code?

Reply:
In the paper it is called "The Transformed BVN Problem" in section 2.2 equation 3, while the original is called "The Standard BVN Problem" in section 2.1 equation 1.  The transformation is that the double-integral is transformed into a single integral with the substitution `r = sin(theta)`.  I renamed the function `transformed_bvn`, is that suitable?